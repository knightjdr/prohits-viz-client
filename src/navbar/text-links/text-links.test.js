import React from 'react';
import { cleanup, render } from '@testing-library/react';

import TextLinks from './text-links';

jest.mock('./text-links-style');

describe('Navbar text links', () => {
  let container;
  let getByText;

  afterAll(() => {
    cleanup();
  });

  beforeAll(() => {
    const links = ['analysis', 'help'];
    const route = 'analysis';
    ({ container, getByText } = render(
      <TextLinks
        links={links}
        route={route}
      />,
    ));
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render analysis link', () => {
    expect(getByText(/analysis/i)).toBeInTheDocument();
  });

  it('should render help link', () => {
    expect(getByText(/help/i)).toBeInTheDocument();
  });
});
