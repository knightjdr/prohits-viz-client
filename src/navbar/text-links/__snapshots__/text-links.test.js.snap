// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar text links should match snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:link {
  color: #2468e5;
}

.c0:visited {
  color: #2468e5;
}

.c0:hover,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus {
  outline: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <ul>
    <li
      class="nav__link_active"
    >
      <a
        class="c0 nav__link"
        href="/analysis"
      >
        analysis
      </a>
    </li>
    <li>
      <a
        class="c0 nav__link"
        href="/help"
      >
        help
      </a>
    </li>
  </ul>
</div>
`;
